### ISSUE TITLE
Warning about "missing @Override" only shown for anonymous inner classes

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#706]
My workspace is configured to mark a "Missing `@Override` annotation" with a Warning, even for interface methods.

Consider this interface:

```java
package test;
public interface I {
	void myMethod(String a, int b);
}
```

and this Groovy class:

```groovy
package test
class A implements I {
	public void myMethod(String a, int b) {
		def i = new I() {
			public void myMethod(String a1, int b1) {
			}
		}
	}
}
```

You can see that `A.myMethod(String, int)` declaration is **not** marked with a warning, while `new I().myMethod(String, int)` is.

**Comment by eric-milles** [comment]
At the moment, the difference exists because the anonymous inner types go through an extra resolve step to help with all the refactoring bugs.  The error/warning is generated from `org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.resolveStatements()`

**Comment by eric-milles** [comment]
Ready to test

**Comment by mauromol** [comment]
Verified in 3.2.0.xx-201810230255-e48, thank you 👍

**Comment by mauromol** [comment]
Eric please note that this bug appears in https://github.com/groovy/groovy-eclipse/wiki/3.1.0-Release-Notes, while it should not (since its milestone is 3.2.0 and not 3.1.0).

**Comment by eric-milles** [comment]
The warning was suppressed for 3.1.0 and is now supported for all methods in 3.2.0

**Comment by mauromol** [comment]
Indeed, so I think this fix should not be in 3.1.0 release notes, should it?
